tests/codegen/builtin_arg.cpp

```diff
@@ -14,14 +14,14 @@ declare i64 @llvm.bpf.pseudo(i64, i64) #0
 ; Function Attrs: argmemonly nounwind
 declare void @llvm.lifetime.start.p0i8(i64, i8* nocapture) #1
 
-define i64 @"kprobe:f"(i8* nocapture readonly) local_unnamed_addr section "s_kprobe:f_1" {
+define i64 @"kprobe:f"(i8*) local_unnamed_addr section "s_kprobe:f_1" {
```

Comment:
Does the issue still exist if this changes from i8* to i64*? Marking it volatile feels a bit hacky.

Reply:
Yes, I tried it but the result was the same. I also tried to use #1105, but it didn't help.

bcc uses volatile load for some register accesses (iovisor/bcc#1212). I can reproduce this error using bcc, so I created the issue iovisor/bcc#2745.

Reply:
As of volatile context access, yonghong-song said:

> Using the volatile to workaround the issue is the right way to do at this point. The issue is fixed in llvm trunk (llvm 11.0): llvm/llvm-project@d96c1bb#diff-6e02af2c61254c488f583a59982eb684
> https://github.com/iovisor/bcc/issues/2745#issuecomment-583757684

llvm 11.0 just disables ReduceLoadWidth optimization.